contracts/liquidity_pool.cairo

```diff
@@ -0,0 +1,85 @@
+%lang starknet
+
+# Part of the main contract to not add complexity by having to transfer tokens between our own contracts
+from lptoken import mint, totalSupply
+
+from Math64x61 import (  # prefix contracts because code is in lib/cairo_math_64x61/**contracts**/Math64x61.cairo
+    Math64x61_fromFelt,
+    Math64x61_mul,
+    Math64x61_div,
+    Math64x61_add,
+    Math64x61_sub,
+    Math64x61_min,
+    Math64x61_ONE,
+    Math64x61_fromUint256,
+)
+from starkware.cairo.common.uint256 import (
+    Uint256,
+    uint256_unsigned_div_rem
+)
+from starkware.starknet.common.syscalls import (
+    get_caller_address,
+    get_contract_address,
+)
+from openzeppelin.token.erc20.interfaces import IERC20
+
+
+
+# # lptoken_amt * exchange_rate = underlying_amt
+# # @returns Math64x61 fp num
+# func get_lptoken_exchange_rate{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+#     pooled_token_addr: felt
+# ) -> (exchange_rate: felt):
+#     let (lpt_supply) = totalSupply()
+#     let (own_addr) = get_contract_address()
+#     let (reserves) = IERC20.balanceOf(contract_address=pooled_token_addr, account=own_addr)
+#     let (exchange_rate) = Math64x61_div(lpt_supply, reserves)
+#     return (exchange_rate)
+# end
+
+func get_lptokens_for_underlying{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    pooled_token_addr: felt,
+    underlying_amt: Uint256
+) -> (lpt_amt: Uint256):
+    alloc_locals
+    tempvar reserves = IERC20.balanceOf(contract_address=pooled_token_addr, account=own_addr)
+
+    if reserves.low == 0:
+        return (underlying_amt)
+    end
+    let (lpt_supply) = totalSupply()
```

Comment:
could the `totalSupply` be called something like `lpTokenTotalSupply` since the option tokens will have the same

also we are mixing camel case and snake case... nothing to do with it, I'm just saying its pain :)